Configuration/PyReleaseValidation/python/relval_steps.py

```diff
@@ -1745,8 +1745,10 @@ def lhegensim2018ml(fragment,howMuch):
 
 steps['DIGIPRMXUP15_PU50']=merge([digiPremixUp2015Defaults50ns])
 steps['DIGIPRMXUP17_PU25']=merge([digiPremixUp2017Defaults25ns])
+steps['DIGIPRMXUP17_PU25_RD']=merge([digiPremixUp2017Defaults25ns, { '--customise_commands':"\"process.EcalLaserCorrectionServiceMC = cms.ESProducer('EcalLaserCorrectionServiceMC') \\n process.GlobalTag.toGet = cms.VPSet( cms.PSet(     record = cms.string('EcalLaserAPDPNRatiosMCRcd'),  tag = cms.string('EcalLaserAPDPNRatios_UL_2017_mc'),  connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS') ) )  \\n process.ecal_sim_parameter_map.timeDependent=cms.bool(True)\"" } ])
```

Comment:
Don't you also need to specify a run number from the 2017 era to be used in the simulation? Otherwise how do you get the laser corrections from the appropriate IOV?

Reply:
Currently this is a test, and it is using only 1 IOV.

As shown in the slides linked, in order to have the time dependent I did, for my private production, the following:

  ```
  setRunNumberForEachLumi = cms.untracked.vuint32(
                                                    298475,
                                                    298486, 
                                                    298615,
                                                    303454
                                                   ),  

``` 

in the digipremixing, and

``` 
numberEventsInLuminosityBlock = cms.untracked.uint32(250)
``` 

in the sim step.

Would you like me to prepare the workflow as this?
Is this how things will be produced centrally?

Reply:
Don't you also need to include a time-dependent `EcalLaserAPDPNRatiosRcd` in these tests? For example, I would have expected you to need to use something like: [EcalLaserAPDPNRatios_Run_Dep_MC](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/tags/EcalLaserAPDPNRatios_Run_Dep_MC). For example, for a ten-event relval for a test covering five 2018 IOVs, you would need in the GEN step:
```
process.source.numberEventsInLuminosityBlock = cms.untracked.uint32(2)
```
and in the DIGIPRMX step:
```
process.source.setRunNumberForEachLumi =  cms.untracked.vuint32(316082, 316720, 317527, 320917, 321414)
process.GlobalTag.toGet = cms.VPSet(
  cms.PSet(record = cms.string("EcalLaserAPDPNRatiosRcd"),
           tag = cms.string("EcalLaserAPDPNRatios_Run_Dep_MC_cleaned"),
           connect = cms.string("frontier://FrontierProd/CMS_CONDITIONS")
          ),
  cms.PSet(record = cms.string("EcalLaserAPDPNRatiosMCRcd"),
           tag = cms.string("EcalLaserAPDPNRatios_UL_2018_mc_3sigma_v2"),
           connect = cms.string("frontier://FrontierProd/CMS_CONDITIONS")
          )
)
```
I can't comment on the plans for central production but I think that the relval workflow should test the use of multi-IOV tags in MC. As far as I can tell, it does not. Or have I misunderstood?

Reply:
> Don't you also need to include a time-dependent `EcalLaserAPDPNRatiosRcd` in these tests? For example, I would have expected you to need to use something like: [EcalLaserAPDPNRatios_Run_Dep_MC](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/tags/EcalLaserAPDPNRatios_Run_Dep_MC). For example, for a ten-event relval for a test covering five 2018 IOVs, you would need in the GEN step:
> 
> ```
> process.source.numberEventsInLuminosityBlock = cms.untracked.uint32(2)
> ```
> 
> and in the DIGIPRMX step:
> 
> ```
> process.source.setRunNumberForEachLumi =  cms.untracked.vuint32(316082, 316720, 317527, 320917, 321414)
> process.GlobalTag.toGet = cms.VPSet(
>   cms.PSet(record = cms.string("EcalLaserAPDPNRatiosRcd"),
>            tag = cms.string("EcalLaserAPDPNRatios_Run_Dep_MC_cleaned"),
>            connect = cms.string("frontier://FrontierProd/CMS_CONDITIONS")
>           ),
>   cms.PSet(record = cms.string("EcalLaserAPDPNRatiosMCRcd"),
>            tag = cms.string("EcalLaserAPDPNRatios_UL_2018_mc_3sigma_v2"),
>            connect = cms.string("frontier://FrontierProd/CMS_CONDITIONS")
>           )
> )
> ```
> 
> I can't comment on the plans for central production but I think that the relval workflow should test the use of multi-IOV tags in MC. As far as I can tell, it does not. Or have I misunderstood?

ok, I will prepare the worflow to use more IOVs, meaning I will prepare the python configuration, but the c++ code here in this PR is independent from it.

Question to relval experts: where is the number of events per workflow defined? Is it handled with some configuration?
At the end we will ask for a "big relval" sample of Zee to estimate the performance of the time dependent MC, and since we have a fixed number of IOVs we can handle, I need to set properly the "numberEventsInLuminosityBlock".
Should this be part of this PR?

Reply:
@amassiro : sorry I have missed your question. The number of events per workflow is set here (https://github.com/cms-sw/cmssw/blob/7d89f4d3f0c521ac8337867424cd0089451b9775/Configuration/PyReleaseValidation/python/relval_steps.py#L589). Kby(9,50) means 9x1000 event will be produced and 50 events per job. By default we set it to 9k events by default for normal statistics. If you need "big relval", you can change these two values properly.

For "numberEventsInLuminosityBlock", you can add this parameter similarly to this line https://github.com/cms-sw/cmssw/blob/7d89f4d3f0c521ac8337867424cd0089451b9775/Configuration/PyReleaseValidation/python/relval_steps.py#L1507

Chayanit for PdmV